app/controllers/comments_controller.rb

```diff
@@ -1,5 +1,5 @@
 class CommentsController < ApplicationController
-  before_action :logged_in_user, only: [:create, :destroy]
+  before_action :user_signed_in?, only: [:create, :destroy]
```

Comment:
:user_signed_in? method này define ở đâu em

Reply:
Dạ cái này Gem Devise nó define hộ mình ạ. Medthod này để kiểm tra xem đã login chưa. Ngoài ra còn một số Method khác Gem Devise define sẵn cho mình ví dụ:
current_user: Người dùng hiện tại
user_session: Phiên truy cập hiện tại của người dùng
sign_in(@user) , sign_out(@user) : đăng nhập, đăng xuất.

Reply:
Em nhầm cái method, method đúng để check login là authenticate_user, hàm này kiểm tra login in chưa và xử lý nếu chưa login thì  đẩy ra trang login kèm thông báo. Còn cái user_signed_in? là kiểm tra login chưa trả về true false( dùng trong view). Em nhầm nhọt xíu :))))